packages/sdk/src/methods/v2Pairs.ts

```diff
@@ -32,6 +33,11 @@ export async function v2Pairs(
   currencies: Array<[Currency, Currency]>
 ): Promise<[PairState, Pair | null][]> {
   const tokens = currencies.map(([currencyA, currencyB]) => [currencyA.wrapped, currencyB.wrapped]);
+  let pairDayData = JSON.parse(localStorage.getItem("voltage_latest_pair_data"))?.data;
```

Comment:
The reason I did it on gooddapp side was for two main reasons:
1. I don't know in this flow directly when a user has entered the swap page
2. passing down pair-data from anywhere involved updating 4-5 different methods throughout the sdk-v1
3. I have a version where I added latestUpdate to storage and check if it was 24h ago before last fetch. but considering as you said yourself the pairs will not likely change (much) I removed it

Reply:
i dont see any new changes to this PR

Reply:
Because I didn't know if my explanation on the 'why' changed your perspective?

So the requested changes are:
1. only run on once on page load
2. pass down pairdata from protocolUI > sdk-v1?

Yes/No?

Reply:
1. yes once per page load
2. I dont understand why protocolUI needs to pass anything to sdk besides the user inputs

Reply:
fixed here: https://github.com/GoodDollar/GoodWeb3-Mono/pull/134/commits/fa35a4dabbf7af9b862915886a0d1c11247a09e2

Did add the sample of rpc's back
Did not add fuse nodies rpc back, because during testing I noticed it still denies the requests when trying to determine optimal route